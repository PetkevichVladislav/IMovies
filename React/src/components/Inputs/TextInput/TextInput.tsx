import { FC, useState } from "react";

import "./TextInput.scss";
import "../Input.scss";

export interface ITextInput{
    placeholder: string;
    value?: string;
    height?: string;
    name?: string;  
    onChange?: (changedVlaue?: string) => void;
}

export const TextInput : FC<ITextInput> = ({placeholder, value = "", height, name, onChange}) => {
    return (
        <>
            <input style={{height:`${height}`}} 
            className="input text-input" 
            name={name} 
            placeholder={placeholder} 
            value={value}
            onChange={event => onChange && onChange(event.target.value)}
            data-testid="text-input"/>
        </>
    )
}