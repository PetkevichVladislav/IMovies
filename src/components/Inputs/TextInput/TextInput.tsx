import { FC, useState } from "react";

import "./TextInput.scss";
import "../Input.scss";

export interface ITextInput{
    placeholder: string;
    initialValue?: string|null;
    height?: string;
    name?: string;
}

export const TextInput : FC<ITextInput> = ({placeholder, initialValue, height, name}) => {
    const [value, setValue] = useState(initialValue);
    
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
    };
    
    return (
        <>
            <input style={{height:`${height}`}} className="input text-input" name={name} placeholder={placeholder} onChange={handleChange} value={value!}/>
        </>
    )
}