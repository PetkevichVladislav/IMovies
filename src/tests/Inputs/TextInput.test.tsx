import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { ITextInput, TextInput } from "../../components/Inputs/TextInput/TextInput";

describe("Text Input", () => {
    const textInputData : ITextInput = {
        placeholder: "input your data",
        value: "15",
        name: "data",
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<TextInput {...textInputData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("when input is created it should be properly initialized", () =>{
        render(<TextInput {...textInputData}/>);
        const input = screen.getByTestId("text-input");
        expect(input).toHaveAttribute("name",textInputData.name);
        expect(input).toHaveAttribute("value",textInputData.value);
        expect(input).toHaveAttribute("placeholder",textInputData.placeholder);
    });
});