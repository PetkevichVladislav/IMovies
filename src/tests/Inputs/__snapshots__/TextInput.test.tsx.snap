// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input When data is valid component should renders correctly 1`] = `
<input
  className="input text-input"
  data-testid="text-input"
  name="data"
  onChange={[Function]}
  placeholder="input your data"
  style={
    Object {
      "height": "undefined",
    }
  }
  value="15"
/>
`;
